### DISCUSSION TITLE
What is the recommended way to manage and structure queries that returns a list with the error plugin

### DISCUSSION THREAD

**Discussion by zulhilmizainuddin** [hayes/pothos#1248] [Q&A] [answered]
The error plugin explains on how to manage and structure queries that are capable of returning errors using GraphQL interfaces and unions.

Say I have the following schema with a `user` query. This schema was generated using the error plugin.

```graphql
interface Error {
  message: String!
}

type NotFoundError implements Error {
  message: String!
}

type Query {
  user(id: ID!): UserResult!
}

type User {
  firstname: String!
  id: ID!
  lastname: String!
}

union UserResult = NotFoundError | User
```

What is the recommended way to manage and structure queries that returns a list with the error plugin?
I am trying to figure out what the schema and query would look like.

Let's say I want to add a `users` query to the schema. What should the `users` query look like?

Would the `users` query look something like this? which would only return existing requested users data if IDs of non existing users are also included in the request
```graphql
users(ids: [ID!]!): [User!]!
```

or this? which would return existing requested users data as well as `NotFoundErrors`s for non existing users
```graphql
users:(ids: [ID!]!): [UserResult!]!
```

or some completely different structure?

**Comment by iamchanii** [comment]
It depends on how and what you want to do with it. here is my opinions:

- If I wanna only return existing users data, I'll: `[User!]!`
- If I care total count of data, but also want to return existing user data, I'll: `[User]!`

Of course, `[UserResult!]!` is maybe an option. but as far as I know, the Error plugins will be not working like that. if you had set error option on query field, the type which created will be `[User!]! | QueryUsersResult!`.

Disclaimer; I've never set error options for query fields. If I'm wrong, please let me know.

**Comment by hayes** [comment, marked as answer]
You can see an example of how list fields work here: https://stackblitz.com/edit/typescript-2ptbbh?file=index.ts

The `directResult` option doesn't work on list fields, so a Result type with a data field containing the list will be created instead.
